File: dart/RegExpBuilderFactory.dart
#import('RegExpBuilder.dart');

class RegExpBuilderFactory {
  
  RegExpBuilder ignoreCase() {
    return new RegExpBuilder().ignoreCase();
  }
  
  RegExpBuilder multiLine() {
    return new RegExpBuilder().multiLine();
  }
  
  RegExpBuilder startOfInput() {
    return new RegExpBuilder().startOfInput();
  }
  
  RegExpBuilder startOfLine() {
    return new RegExpBuilder().startOfLine();
  }
  
  RegExpBuilder endOfInput() {
    return new RegExpBuilder().endOfInput();
  }
  
  RegExpBuilder endOfLine() {
    return new RegExpBuilder().endOfLine();
  }
  
  RegExpBuilder either(Dynamic r) {
    return new RegExpBuilder().either(r);
  }
  
  RegExpBuilder exactly(int n) {
    return new RegExpBuilder().exactly(n);
  }
  
  RegExpBuilder min(int n) {
    return new RegExpBuilder().min(n);
  }
  
  RegExpBuilder max(int n) {
    return new RegExpBuilder().max(n);
  }
  
  RegExpBuilder ahead(RegExpBuilder r) {
    return new RegExpBuilder().ahead(r);
  }
  
  RegExpBuilder notAhead(RegExpBuilder r) {
    return new RegExpBuilder().notAhead(r);
  }
  
  RegExpBuilder then(String s) {
    return new RegExpBuilder().then(s);
  }
  
  RegExpBuilder find(String s) {
    return new RegExpBuilder().find(s);
  }
  
  RegExpBuilder some(List<String> s) {
    return new RegExpBuilder().some(s);
  }
  
  RegExpBuilder maybeSome(List<String> s) {
    return new RegExpBuilder().maybeSome(s);
  }
  
  RegExpBuilder maybe(String s) {
    return new RegExpBuilder().maybe(s);
  }
  
  RegExpBuilder something() {
    return new RegExpBuilder().something();
  }
  
  RegExpBuilder somethingBut(String s) {
    return new RegExpBuilder().somethingBut(s);
  }
  
  RegExpBuilder anything() {
    return new RegExpBuilder().anything();
  }
  
  RegExpBuilder anythingBut(String s) {
    return new RegExpBuilder().anythingBut(s);
  }
  
  RegExpBuilder any() {
    return new RegExpBuilder().any();
  }
  
  RegExpBuilder lineBreak() {
    return new RegExpBuilder().lineBreak();
  }
  
  RegExpBuilder whitespace() {
    return new RegExpBuilder().whitespace();
  }
  
  RegExpBuilder notWhitespace() {
    return new RegExpBuilder().notWhitespace();
  }
  
  RegExpBuilder tab() {
    return new RegExpBuilder().tab();
  }
  
  RegExpBuilder digit() {
    return new RegExpBuilder().digit();
  }
  
  RegExpBuilder notDigit() {
    return new RegExpBuilder().notDigit();
  }
  
  RegExpBuilder letter() {
    return new RegExpBuilder().letter();
  }
  
  RegExpBuilder notLetter() {
    return new RegExpBuilder().notLetter();
  }
  
  RegExpBuilder lowerCaseLetter() {
    return new RegExpBuilder().lowerCaseLetter();
  }
  
  RegExpBuilder upperCaseLetter() {
    return new RegExpBuilder().upperCaseLetter();
  }
  
  RegExpBuilder append(RegExpBuilder r) {
    return new RegExpBuilder().append(r);
  }
  
  RegExpBuilder optional(RegExpBuilder r) {
    return new RegExpBuilder().optional(r);
  }
}

File: python/regexpbuilder/RegExpBuilderFactory.py
from RegExpBuilder import RegExpBuilder

class RegExpBuilderFactory:
  
    def ignore_case(self):
        return RegExpBuilder().ignore_case()
  
    def multi_line(self):
        return RegExpBuilder().multi_line()
  
    def start_of_input(self):
        return RegExpBuilder().start_of_input()

    def start_of_line(self):
        return RegExpBuilder().start_of_line()
  
    def end_of_input(self):
        return RegExpBuilder().end_of_input()

    def end_of_line(self):
        return RegExpBuilder().end_of_line()

    def either(self, s):
        return RegExpBuilder().either(s)
  
    def exactly(self, n):
        return RegExpBuilder().exactly(n)
  
    def min(self, n):
        return RegExpBuilder().min(n)
  
    def max(self, n):
        return RegExpBuilder().max(n)
  
    def ahead(self, r):
        return RegExpBuilder().ahead(r)
  
    def not_ahead(self, r):
        return RegExpBuilder().not_ahead(r)

    def then(self, s):
        return RegExpBuilder().then(s)

    def find(self, s):
        return RegExpBuilder().find(s)

    def some(self, s):
        return RegExpBuilder().some(s)

    def maybe_some(self, s):
        return RegExpBuilder().maybe_some(s)

    def maybe(self, s):
        return RegExpBuilder().maybe(s)

    def something(self):
        return RegExpBuilder().something()

    def something_but(self, s):
        return RegExpBuilder().something_but(s)

    def anything(self):
        return self.min(0).of_any()

    def anything_but(self, s):
        return RegExpBuilder().anything_but(s)

    def any(self):
        return RegExpBuilder().any()

    def line_break(self):
        return RegExpBuilder().line_break()

    def whitespace(self):
        return RegExpBuilder().whitespace()

    def not_whitespace(self):
        return RegExpBuilder().not_whitespace()

    def tab(self):
        return RegExpBuilder().tab()

    def digit(self):
        return RegExpBuilder().digit()

    def not_digit(self):
        return RegExpBuilder().not_digit()

    def letter(self):
        return RegExpBuilder().letter()

    def not_letter(self):
        return RegExpBuilder().not_letter()

    def lower_case_letter(self):
        return RegExpBuilder().lower_case_letter()

    def upper_case_letter(self):
        return RegExpBuilder().upper_case_letter()

    def append(self, r):
        return RegExpBuilder().append(r)

    def optional(self, r):
        return RegExpBuilder().optional(r)
